packages/fxa-settings/src/pages/Signin/Signin/index.tsx

```diff
@@ -0,0 +1,201 @@
+  serviceName,
+}: SigninProps & RouteComponentProps) => {
+  usePageViewEvent('signin', {
+    entrypoint_variation: 'react',
+  });
+
+  const ftlMsgResolver = useFtlMsgResolver();
+  const localizedPasswordFormLabel = ftlMsgResolver.getMsg(
+    'password',
+    'Password'
+  );
+  // TODO: This page is also supposed to render the user card, complete with avatar.
+  // The avatar mixin is complex enough that I think it deserves its own ticket.
+
+  // TODO: The user-card mixin does some fancy footwork listening to see if you access token changes
+  // because it's possible for the access token to be invalidated when we display the avatar image.
+  // if we get the notification of a change, we re-render the card.
+
+  const onSubmit = useCallback(async () => {
+    try {
+      // TODO: add logic to set a new password
+      // TODO: add message in Alert Bar re: success, then navigate to where appropriate
+    } catch (e) {
+      // TODO: metrics event for error
+      // TODO: Add in localized Alert Bar message for error
+      setError(e);
+    }
+  }, []);
+
+  const [error, setError] = useState('');
+  const [password, setPassword] = useState('');
+  const useLoggedIn = () => {};
+
+  const signInWithPassword = () => {};
+
+  // TODO:
+  // Add in the Alert Bar in place of the original `success` and `error` display divs
+
+  return (
+    <>
+      <div className="mb-4">
+        {isPasswordNeeded ? (
+          <FtlMsg id="signin-password-needed-header">
+            <h1 className="card-header">
+              Enter your password{' '}
+              <span className="card-subheader">for your Firefox account</span>
+            </h1>
+          </FtlMsg>
+        ) : (
+          <>
+            <FtlMsg id="signin">
+              <h1 className="card-header">Sign in</h1>
+            </FtlMsg>
+            <>
+              {serviceLogo ? (
+                <span className="card-subheader">
+                  <FtlMsg id="signin-subheader-with-logo">Continue to </FtlMsg>
+                  {serviceLogo}
+                </span>
+              ) : (
+                <FtlMsg
+                  id="signin-subheader-without-logo"
+                  vars={{ serviceName }}
+                >
+                  <span className="card-subheader">
+                    Continue to {serviceName}
+                  </span>
+                </FtlMsg>
+              )}
+            </>
+          </>
+        )}
+      </div>
+      <section>
+        {/* Alerts and success messages originally went here */}
+        <div className="mt-9">
+          <div className="avatar-wrapper mx-auto h-24 w-24 mobileLandscape:h-40 mobileLandscape:w-40"></div>
+          <div className="my-5 text-base break-all">{email}</div>
+        </div>
+        <form noValidate {...{ onSubmit }}>
+          <input type="email" className="email hidden" value={email} disabled />
+
+          {isPasswordNeeded && (
+            <InputPassword
+              className="mb-2"
```

Comment:
Maybe adding `text-start` here would be good to align the label, otherwise its indented when we click into the input

Reply:
Oo perfect thank you!